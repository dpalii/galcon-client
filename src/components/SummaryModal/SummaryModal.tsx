import './SummaryModal.css'
import { User } from '../../types'

export interface SummaryModalProps {
    winner: User
    close: () => void
}

export function SummaryModal({ winner, close }: SummaryModalProps) {
    return (
        <div className="backdrop" onClick={() => close()}>
            <div className="modal" onClick={(e) => e.stopPropagation()}>
                <h2 className="heading">
                    {`${winner.name} WON`}
                </h2>
                <button className="btn" onClick={() => close()}>
                    CLOSE
                </button>
            </div>
        </div>
    )
}
