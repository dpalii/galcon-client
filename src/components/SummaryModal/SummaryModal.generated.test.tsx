/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/no-unnecessary-act */
import { render } from "react-dom";
import { fireEvent} from '@testing-library/react'

import { SummaryModal } from '../../../src/components/SummaryModal/SummaryModal';

import { unmountComponentAtNode } from "react-dom";
import { act } from 'react-dom/test-utils';

let container: any = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('<SummaryModal>', () => {
  it('should render component', () => {
    act(() => {
      render(<SummaryModal  
        winner={{name:'winner', color: '#000000', id: '2', isHost: true}}  
        close={ () => {} } />, container);
    });
    container.querySelector('.backdrop').dispatchEvent(new MouseEvent("click", { bubbles: true }));
    container.querySelector('.btn').dispatchEvent(new MouseEvent("click", { bubbles: true }));
    expect(container).toMatchSnapshot();
  })
});