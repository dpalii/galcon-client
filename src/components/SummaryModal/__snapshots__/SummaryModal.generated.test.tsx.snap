// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SummaryModal> should render component 1`] = `
<div>
  <div
    class="backdrop"
  >
    <div
      class="modal"
    >
      <h2
        class="heading"
      >
        winner WON
      </h2>
      <button
        class="btn"
      >
        CLOSE
      </button>
    </div>
  </div>
</div>
`;
